modules/SofaGeneralAnimationLoop/src/SofaGeneralAnimationLoop/MechanicalMatrixMapper.inl

```diff
@@ -381,17 +381,19 @@ void MechanicalMatrixMapper<DataTypes1, DataTypes2>::addKToMatrix(const Mechanic
     sofa::core::MultiMatrixDerivId c = sofa::core::MatrixDerivId::mappingJacobian();
     const MatrixDeriv1 &J1 = c[ms1].read()->getValue();
     const MatrixDeriv2 &J2 = c[ms2].read()->getValue();
-
+    if (this->getContext()->getTime() == 0)
+        nbColsJ1 = J1.begin().row().size()*DerivSize1;
     Eigen::SparseMatrix<double> J1eig;
     Eigen::SparseMatrix<double> J2eig;
-    J1eig.resize(K->nRow, J1.begin().row().size()*DerivSize1);
-    unsigned int nbColsJ1 = 0, nbColsJ2 = 0;
+    J1eig.resize(K->nRow, nbColsJ1);
```

Comment:
Ideally yes, but in practice, it is not easy, because I need the mparams and the matrixAccessor inputs from addKtoMatrix to set up the dimension. I made a new commit to improve that version a little bit.

Reply:
Oh wait, my bad, it's not a problem actually, I will commit a better version in a minute